pkg/controller/registration/registration_controller.go

```diff
@@ -158,12 +162,32 @@ func isServiceAccountKeyCRD(crd *apiextensions.CustomResourceDefinition) bool {
 	return crd.Spec.Group == serviceAccountKeyAPIGroup && crd.Spec.Names.Kind == serviceAccountKeyKind
 }
 
-func RegisterDefaultController(r *ReconcileRegistration, crd *apiextensions.CustomResourceDefinition, gvk schema.GroupVersionKind) (k8s.SchemaReferenceUpdater, error) {
+func RegisterDefaultController(config *controller.Config) registrationFunc { //nolint:revive
+	return func(r *ReconcileRegistration, crd *apiextensions.CustomResourceDefinition, gvk schema.GroupVersionKind) (k8s.SchemaReferenceUpdater, error) {
+		return registerDefaultController(r, config, crd, gvk)
+	}
+}
+
+func registerDefaultController(r *ReconcileRegistration, config *controller.Config, crd *apiextensions.CustomResourceDefinition, gvk schema.GroupVersionKind) (k8s.SchemaReferenceUpdater, error) {
 	if _, ok := k8s.IgnoredKindList[crd.Spec.Names.Kind]; ok {
 		return nil, nil
 	}
-	// Depending on which resource it is, we need to register a different controller.
+
 	var schemaUpdater k8s.SchemaReferenceUpdater
+
+	if kccfeatureflags.UseDirectReconciler(gvk.GroupKind()) {
+		switch gvk.GroupKind() {
+		case schema.GroupKind{Group: "apikeys.cnrm.cloud.google.com", Kind: "APIKeysKey"}:
+			if err := apikeys.AddKeyReconciler(r.mgr, config); err != nil {
+				return nil, err
+			}
+			return schemaUpdater, nil
+		default:
+			klog.Warningf("requested direct reconciler for %v, but it is not supported", gvk.GroupKind())
```

Comment:
This at a minimum needs to be an Error. Worried that we could fall through to an alternate reconciler. Worried about that reconciler then doing something we don't want with the customers objects.

Reply:
Hmmm, good point, I'll take a look!